{
synchronized (lock) {
    try {
      target.trigger(timestamp);
    }
 catch (    Throwable t) {
      LOG.error("Caught exception while processing timer.",t);
      if (task.timerException == null) {
        task.timerException=new TimerException(t);
      }
    }
  }
}
